frontend-react/src/hooks/UseLookupTable.ts

```diff
@@ -91,14 +90,54 @@ export const getSenderAutomationData = async <T>(
     );
 };
 
-const useLookupTable = <T>(tableName: LookupTables): ValueSet[] => {
-    const [valueSetArray, setValueSetArray] = useState<ValueSet[]>([]);
+export const getSenderAutomationDataRows = async <T>(
+    tableName: LookupTables,
+    dataSetName: string | null = null
+): Promise<any[]> => {
+    const version: number = await getLatestVersion(tableName);
+    if (version === undefined) {
+        showError("DANGER! no version was found");
```

Comment:
Oh sweet, I'll work this into the api stuff so maybe we can automate some of the toasts when server errors happen. That said, also making an effort to focus on logging for developer purposes. A good `console.error` or try/catch/throw would be great. It always helps to know any reason _why_ latest version is undefined.

Reply:
@kevinhaube i just refactored the error handling a bit. The error in question here doesn't actually throw any exception- it's almost like an application error, except that the real root cause would be that the lookuptable wasn't loaded. Regardless, after thinking about it, I felt like that check should be consolidated within the GetLatestVersion method anyway- which was already doing all the try/catch stuff you asked for. Lmk what you think of that refactor- i'm not 100% satisfied with it, but then I'm never 100% satisfied with how error handling is structured. Ever.

Reply:
It looks good to me! Agreed, I'm trying to inch towards a well-crafted error handling layer. The main thing I was looking for were the console calls, but thanks for the try/catch addition too! That's a pattern I've found super helpful in the work I'm on right now and I think can help us pipe errors up to components a lot easier, which is ultimately where we need them.